### DISCUSSION TITLE
Adding documents to an already existing graph

### DISCUSSION THREAD

**Discussion by jonnyAkar** [microsoft/graphrag#982] [General]
Hi guys, I was wondering if it's possible to add chunks to an already existing graph? For example if i want to make a graphRag of my emails, it would be very inefficient to rerun the entire process every time i get an email. Is there a way to have an already existing graph that I could then add documents to? Thanks!

**Comment by NathanAP** [comment]
I'm not an expert in GraphRAG but I've been using it since 2 or 3 weeks now. I believe I also had this problem/idea on how to add new data to a graph or how to override content on a graph. 

It need to rerun everything to be able to figure which chunks (content) are linked and create new relationships, entities, etc. Also, if your document is already indexed, it will be in the cache o GraphRag, so that won't be re-indexed. Not sure if this answers anything for you...

**Comment by antopiahk** [comment]
The workaround I'm using until a more official implementation is made using a short term format, like using a vector DB, or for your case even providing the new emails in the prompt itself works without adding too much to the cost. if you use a vector DB, RAG on that to then access relevant parts works. Then you can schedule regular indexing using the new data. Can't wait until they implement re-indexing!
